### PR TITLE
Rescue TypeError During PEheader Processing

### PR DESCRIPTION
**Pull request by jerryk55**
Currently TypeError is ignored during processing of the .ico files in the PEheader processing code.
This results in all the information collected about all files in the VM being discarded and no new
info being available about said files.

Rescue the TypeError so that processing of the files can continue successfully.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1805467

This can be backported as far back as we'd like.

@roliveri please review and merge when appropriate.

### CODE REVIEW

**Comment by jerryk55** [comment]
git add_label smartstate

**Comment by jerryk55** [comment]
Note that all the RuboCop complaints are either against existing code that was indented or $log.debug statements in quotes, which are consistent with the other $log calls in this file.  As such they will be ignored.  Looking into the codeclimate complaints now...

File: lib/metadata/util/md5deep.rb [hunk 1]
```diff
@@ -139,12 +139,15 @@ def processFile(path, x, xmlNode)
...
           if @opts.winVerList.include?(ext)
-            if @opts.versioninfo || @opts.imports
+            begin
               peHdr = PEheader.new(fh) rescue nil
               unless peHdr.nil?
                 xmlFileNode.add_element("versioninfo", peHdr.versioninfo) if @opts.versioninfo && !peHdr.versioninfo.blank?
                 xmlFileNode.add_element("libraries", "imports" => peHdr.getImportList) if @opts.imports && !peHdr.imports.blank?
               end
+            rescue TypeError => err
+              $log.debug "processFile: TypeError handling PEheader; skipping PEheader info"
```
Comment (roliveri) [inline lib/metadata/util/md5deep.rb]: Should this be debug or info? Will we want to see that this happened in a non-debug log?
Maybe just the backtrace part should be debug.

Comment (jerryk55) [reply lib/metadata/util/md5deep.rb]: @roliveri I can do that.  A lot easier than dealing with codeclimate nesting complaints.

**Comment by jerryk55** [comment]
@roliveri I still need to test that the version and library info is still being collected correctly after these changes before this is merged - so hold off on doing so for now after your review.  Thanks.

**Comment by jerryk55** [comment]
@roliveri this is ready for final review and merge when you can.  As far as back-porting - I'll leave that decision to you and @dmetzger57

**Comment by roliveri** [comment]
Released version "0.5.2" of manageiq-smartstate.

**Comment by simaishi** [comment]
@roliveri Ivanchuk branch is on 0.3.x. Can you please release a new 0.3.x gem with this change?

@jerryk55 Once new version is released, please bump minimum version in core Gemfile for Ivanchuk branch.

**Comment by roliveri** [comment]
Released version 0.3.5

**Pull request merged**